src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_tasklet.py

```diff
@@ -167,16 +171,281 @@ def _visit_deref(self, node: gtir.FunCall) -> MemletExpr | ValueExpr:
                 return MemletExpr(it.field, field_subset)
 
             else:
-                raise NotImplementedError
+                # we use a tasklet to perform dereferencing of a generic iterator
+                assert all(dim in it.indices for dim in it.dimensions)
+                field_indices = [(dim, it.indices[dim]) for dim in it.dimensions]
+                index_connectors = [
+                    IndexConnectorFmt.format(dim=dim.value)
+                    for dim, index in field_indices
+                    if not isinstance(index, SymbolExpr)
+                ]
+                index_internals = ",".join(
```

Comment:
Just an idea, but you should include "literal" or something like this in the name to make clear that the index are not really runtime dependent (which is not true, since scalar inputs are turned into a symbol and could be used here). But I think you get my point.
However, feel free to ignore.